### PR TITLE
Fixes various element inspector bugs

### LINKED ISSUE #668: If you set function to decreasing the default should be denied
- [x] Default of decreasing value should be denied
- [x] Functions don't save the value if you change it (ex. if you set the end value to partially denied it resets to partially satisfied)
- [x] if you choose stochastic initial sat value should switch to (no value)

Screen Shot 2022-06-22 at 3 52 31 PM

### PR DESCRIPTION
**Pull request by joeyels**
Closes #668

### CODE REVIEW

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 1]
```diff
@@ -361,6 +361,12 @@ var ElementInspector = Backbone.View.extend({
                 this.$('#init-sat-value').prop('disabled', false);
             }
         }
+
+        //Changes initial satisfaction value to no value for stochastic and stochastic constant functions
+        if (functionType == 'RC') {
+            console.log("test");
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Remove console logs?

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 2]
```diff
@@ -360,6 +361,12 @@ var ElementInspector = Backbone.View.extend({
                 this.$('#init-sat-value').prop('disabled', false);
             }
         }
+
+        //Changes initial satisfaction value to no value for stochastic and stochastic constant functions
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js:365]: Add stochastic and constant stochastic for `no value` as well

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 3]
```diff
@@ -361,6 +361,12 @@ var ElementInspector = Backbone.View.extend({
...
+        if (functionType == 'RC') {
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Add stochastic for no value as well

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 4]
```diff
@@ -704,20 +710,19 @@ var ElementInspector = Backbone.View.extend({
...
         }
         // Only creates the FunctionSegmentView if there is a function segment
         if (this.intention.get('evolvingFunction') != null && this.intention.get('evolvingFunction').get('type') != 'NT') {
-
             var funcSegList = this.intention.getFuncSegments();
             var i = 0;
             // Creates a FuncSegView for each of the function segment in the functionSegList
             funcSegList.forEach(
-                funcSeg => {
+                funcSeg => {
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete extra tab space

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 5]
```diff
@@ -185,6 +185,7 @@ var ElementInspector = Backbone.View.extend({
                 }
             } else {
                 this.updateHTML();
+                console.log("runs");
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete console log?

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 6]
```diff
@@ -337,6 +337,7 @@ var ElementInspector = Backbone.View.extend({
     updateHTML: function (event) {
         // Check if selected init sat value and functionType pair is illegal
         // Only runs if evolvingFunction is defined and therefore there is a function type
+        console.log("issue");
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete console log?

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 7]
```diff
@@ -384,6 +390,7 @@ var ElementInspector = Backbone.View.extend({
         var funcTypeChanged = event.target.className == 'function-type';
 
         if (functionType !== null && initValue !== null) {
+            console.log("when does this run");
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete console log?

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 8]
```diff
@@ -909,18 +913,15 @@ var FuncSegView = Backbone.View.extend({
         // Disable the function satisfaction dropdown for constant and stochastic functions
         if (this.model.get('type') == 'C' || this.model.get('type') == 'R') {
             this.$('#seg-sat-value').prop('disabled', 'disabled');
-            if (this.intention.get('evolvingFunction').get('type') !== 'MP' && this.intention.get('evolvingFunction').get('type') !== 'MN' && (this.model.get('type') == 'C')) {
+            if (this.intention.get('evolvingFunction').get('type') !== 'MP' && this.intention.get('evolvingFunction').get('type') !== 'MN' && (this.model.get('type') == 'C')) {
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete extra tab space

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 9]
```diff
@@ -909,18 +913,15 @@ var FuncSegView = Backbone.View.extend({
...
                 if (this.intention.get('evolvingFunction').get('type') == 'SD' || this.intention.get('evolvingFunction').get('type') == 'DS') {
                     this.$("#seg-sat-value").val(this.model.get('refEvidencePair'));
-                } else if (!this.hasUD) {
-                    this.$("#seg-sat-value").val(this.initSatValue);
-                    this.model.set('refEvidencePair', this.initSatValue);
                 }
             }
         } else {
-            this.$('#seg-sat-value').prop('disabled', '');
+            this.$('#seg-sat-value').prop('disabled', '');
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js]: Delete extra tab space

File: leaf-ui/rappid-extensions/ElementInspector.js [hunk 10]
```diff
@@ -909,18 +913,15 @@ var FuncSegView = Backbone.View.extend({
...
-                    this.$("#seg-sat-value").val(this.initSatValue);
-                    this.model.set('refEvidencePair', this.initSatValue);
                 }
             }
         } else {
-            this.$('#seg-sat-value').prop('disabled', '');
+            this.$('#seg-sat-value').prop('disabled', '');   
         }
-
+
```
Comment (YesugenB) [inline leaf-ui/rappid-extensions/ElementInspector.js:920]: Delete extra tab space

**Review by YesugenB — APPROVED** [review]
Looks good!

**Comment by YesugenB** [comment]
I think for `stochastic` and `stochastic-constant`, because the sat values are back to `no value`, it would be good for the dropdown menu for the function types to remain small like in the picture until the user changes the sat value away from `no value`. 
Screen Shot 2022-07-05 at 2 58 23 PM

**Comment by YesugenB** [comment]
Also double-check how the chart should update for `no value`, right now it is keeping the original/previous sat value for chart value.

File: leaf-analysis/app.js [hunk 11]
```diff
@@ -4,7 +4,7 @@
 
 // Name of .jar file for BloomingLeaf project must be Blooming.jar
 //var userPath = "/Users/<your user path here>/BloomingLeaf"
-var userPath = "/Users/Joey/Downloads/SURF/BloomingLeaf"
+var userPath = "/Users/judySmith/git/BloomingLeaf"
```
Comment (sonorah) [inline leaf-analysis/app.js:7]: Looks like this was committed accidentally.

**Pull request merged**